src/core.rs

```diff
@@ -69,15 +69,75 @@ pub struct WormholeWelcome {
     pub code: Code,
 }
 
+pub struct MailboxConnection<V: serde::Serialize + Send + Sync + 'static> {
+    config: AppConfig<V>,
+    server: RendezvousServer,
+    pub welcome: Option<String>,
+    pub mailbox: Mailbox,
+    pub code: Code,
+}
+
+impl<V: serde::Serialize + Send + Sync + 'static> MailboxConnection<V> {
+    pub async fn create(config: AppConfig<V>, code_length: usize) -> Result<Self, WormholeError> {
+        Self::create_with_password(
+            config,
+            &wordlist::default_wordlist(code_length).choose_words(),
+        )
+        .await
+    }
+
+    pub async fn create_with_password(
+        config: AppConfig<V>,
+        password: &str,
+    ) -> Result<Self, WormholeError> {
+        let (mut server, welcome) =
+            RendezvousServer::connect(&config.id, &config.rendezvous_url).await?;
+        let (nameplate, mailbox) = server.allocate_claim_open().await?;
+        let code = Code::new(&nameplate, &password);
+
+        Ok(MailboxConnection {
+            config,
+            server,
+            mailbox,
+            code,
+            welcome,
+        })
+    }
+
+    pub async fn connect(
+        config: AppConfig<V>,
+        code: Code,
+        allocate: bool,
+    ) -> Result<Self, WormholeError> {
+        let (mut server, welcome) =
+            RendezvousServer::connect(&config.id, &config.rendezvous_url).await?;
+        let nameplate = code.nameplate();
+        if allocate {
```

Comment:
```suggestion
        if !allocate {
```

The semantics have flipped with the rename

Reply:
Thanks, this should be fixed now and we have a pair of tests to verify that behaviour.

Reply:
As it would change the behaviour I will revert `connect_with_code(...)` to use `expect_claimed_nameplate` and mark it as deprecated. That should allow a smooth transition to the `allocate` argument.